Index: distrib/f12/yum.repos.d/fedora-updates-testing.repo
===================================================================
--- distrib/f12/yum.repos.d/fedora-updates-testing.repo	(revision 0)
+++ distrib/f12/yum.repos.d/fedora-updates-testing.repo	(revision 0)
@@ -0,0 +1,26 @@
+[updates-testing]
+name=Fedora 12 - $basearch - Test Updates
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/updates/testing/12/$basearch/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=updates-testing-f12&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-test-$basearch
+
+[updates-testing-debuginfo]
+name=Fedora 12 - $basearch - Test Updates Debug
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/updates/testing/12/$basearch/debug/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=updates-testing-debug-f12&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-test-$basearch
+
+[updates-testing-source]
+name=Fedora 12 - Test Updates Source
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/updates/testing/12/SRPMS/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=updates-testing-source-f12&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-test-$basearch
Index: distrib/f12/yum.repos.d/fedora-updates-testing.repo-e
===================================================================
--- distrib/f12/yum.repos.d/fedora-updates-testing.repo-e	(revision 0)
+++ distrib/f12/yum.repos.d/fedora-updates-testing.repo-e	(revision 0)
@@ -0,0 +1,26 @@
+[updates-testing]
+name=Fedora 12 - $basearch - Test Updates
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/updates/testing/12/$basearch/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=updates-testing-f12&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-test-$basearch
+
+[updates-testing-debuginfo]
+name=Fedora 12 - $basearch - Test Updates Debug
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/updates/testing/12/$basearch/debug/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=updates-testing-debug-f12&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-test-$basearch
+
+[updates-testing-source]
+name=Fedora 12 - Test Updates Source
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/updates/testing/12/SRPMS/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=updates-testing-source-f12&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-test-$basearch
Index: distrib/f12/yum.repos.d/fedora.repo
===================================================================
--- distrib/f12/yum.repos.d/fedora.repo	(revision 0)
+++ distrib/f12/yum.repos.d/fedora.repo	(revision 0)
@@ -0,0 +1,26 @@
+[fedora]
+name=Fedora 12 - $basearch
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Everything/$basearch/os/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=fedora-12&arch=$basearch
+enabled=1
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-$basearch
+
+[fedora-debuginfo]
+name=Fedora 12 - $basearch - Debug
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Everything/$basearch/debug/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=fedora-debug-12&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-$basearch
+
+[fedora-source]
+name=Fedora 12 - Source
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Everything/source/SRPMS/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=fedora-source-12&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-$basearch
Index: distrib/f12/yum.repos.d/fedora.repo-e
===================================================================
--- distrib/f12/yum.repos.d/fedora.repo-e	(revision 0)
+++ distrib/f12/yum.repos.d/fedora.repo-e	(revision 0)
@@ -0,0 +1,26 @@
+[fedora]
+name=Fedora 12 - $basearch
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Everything/$basearch/os/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=fedora-12&arch=$basearch
+enabled=1
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-$basearch
+
+[fedora-debuginfo]
+name=Fedora 12 - $basearch - Debug
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Everything/$basearch/debug/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=fedora-debug-12&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-$basearch
+
+[fedora-source]
+name=Fedora 12 - Source
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Everything/source/SRPMS/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=fedora-source-12&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-$basearch
Index: distrib/f12/yum.repos.d/fedora-updates.repo
===================================================================
--- distrib/f12/yum.repos.d/fedora-updates.repo	(revision 0)
+++ distrib/f12/yum.repos.d/fedora-updates.repo	(revision 0)
@@ -0,0 +1,26 @@
+[updates]
+name=Fedora 12 - $basearch - Updates
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/updates/12/$basearch/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=updates-released-f12&arch=$basearch
+enabled=1
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-$basearch
+
+[updates-debuginfo]
+name=Fedora 12 - $basearch - Updates - Debug
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/updates/12/$basearch/debug/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=updates-released-debug-f12&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-$basearch
+
+[updates-source]
+name=Fedora 12 - Updates Source
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/updates/12/SRPMS/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=updates-released-source-f12&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-$basearch
Index: distrib/f12/yum.repos.d/fedora-updates.repo-e
===================================================================
--- distrib/f12/yum.repos.d/fedora-updates.repo-e	(revision 0)
+++ distrib/f12/yum.repos.d/fedora-updates.repo-e	(revision 0)
@@ -0,0 +1,26 @@
+[updates]
+name=Fedora 12 - $basearch - Updates
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/updates/12/$basearch/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=updates-released-f12&arch=$basearch
+enabled=1
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-$basearch
+
+[updates-debuginfo]
+name=Fedora 12 - $basearch - Updates - Debug
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/updates/12/$basearch/debug/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=updates-released-debug-f12&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-$basearch
+
+[updates-source]
+name=Fedora 12 - Updates Source
+failovermethod=priority
+#baseurl=http://download.fedoraproject.org/pub/fedora/linux/updates/12/SRPMS/
+mirrorlist=https://mirrors.fedoraproject.org/mirrorlist?repo=updates-released-source-f12&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://download.fedoraproject.org/pub/fedora/linux/releases/12/Fedora/$basearch/os/RPM-GPG-KEY-fedora-$basearch
Index: distrib/f12/pkgs/01
===================================================================
--- distrib/f12/pkgs/01	(revision 0)
+++ distrib/f12/pkgs/01	(revision 0)
@@ -0,0 +1 @@
+glibc
Index: distrib/f12/pkgs/02
===================================================================
--- distrib/f12/pkgs/02	(revision 0)
+++ distrib/f12/pkgs/02	(revision 0)
@@ -0,0 +1,3 @@
+--reinstall
+
+filesystem
Index: distrib/f12/pkgs/03
===================================================================
--- distrib/f12/pkgs/03	(revision 0)
+++ distrib/f12/pkgs/03	(revision 0)
@@ -0,0 +1,2 @@
+coreutils
+initscripts
Index: distrib/f12/yum/yum.conf
===================================================================
--- distrib/f12/yum/yum.conf	(revision 0)
+++ distrib/f12/yum/yum.conf	(revision 0)
@@ -0,0 +1,11 @@
+[main]
+cachedir=@YUMCACHEDIR@
+persistdir=@YUMLIBDIR@
+reposdir=@YUMETCDIR@/yum.repos.d
+debuglevel=1
+logfile=@YUMLOGDIR@/log
+lockfile=@YUMLOCKDIR@/yum.pid
+pkgpolicy=newest
+distroverpkg=fedora-release
+installonlypkgs=
+exactarch=0
Index: distrib/f12/rpmlist.d/00.lst
===================================================================
--- distrib/f12/rpmlist.d/00.lst	(revision 0)
+++ distrib/f12/rpmlist.d/00.lst	(revision 0)
@@ -0,0 +1,11 @@
+glibc-[0-9]*.rpm
+basesystem-*.rpm
+bash-[0-9]*.rpm
+filesystem-*.rpm
+glibc-common-*.rpm
+libgcc-[0-9]*.rpm
+ncurses-base-[0-9]*.rpm
+ncurses-libs-*.rpm
+nss-softokn-freebl-*.rpm
+setup-*.rpm
+tzdata-*.rpm
Index: distrib/Makefile.am
===================================================================
--- distrib/Makefile.am	(revision 2883)
+++ distrib/Makefile.am	(working copy)
@@ -78,6 +78,9 @@
 			   f11/pkgs/01 f11/pkgs/02 f11/pkgs/03 f11/yum/yum.conf \
 			   $(wildcard $(srcdir)/f11/yum.repos.d/*.repo) \
 			   $(wildcard $(srcdir)/f11/rpmlist.d/*.lst) $(wildcard $(srcdir)/f11/rpmlist.d/*.opt) \
+			   f12/pkgs/01 f12/pkgs/02 f12/pkgs/03 f12/yum/yum.conf \
+			   $(wildcard $(srcdir)/f12/yum.repos.d/*.repo) \
+			   $(wildcard $(srcdir)/f12/rpmlist.d/*.lst) $(wildcard $(srcdir)/f12/rpmlist.d/*.opt) \
 			   centos4/pkgs/01 centos4/pkgs/02 centos4/pkgs/03 \
 			   centos4/yum/yum.conf $(wildcard $(srcdir)/centos4/yum.repos.d/*.repo) \
 			   centos5/pkgs/01 centos5/pkgs/02 centos5/pkgs/03 \
@@ -105,7 +108,7 @@
 			   $(nobase_confdistrib_DATA) \
 			   $(defaults_DATA)
 
-redhat_style		=  rh9 fc1 fc2 fc3 fc4 fc5 fc6 f7 f8 f9 f10 f11 centos4 centos5
+redhat_style		=  rh9 fc1 fc2 fc3 fc4 fc5 fc6 f7 f8 f9 f10 f11 f12 centos4 centos5
 sid_style		=  etch lenny squeeze sid
 
 install-exec-hook:	install-notify-xtra
